File: 3291-find-if-array-can-be-sorted/3291-find-if-array-can-be-sorted.cpp
class Solution {
public:
    static int setBitsCount(int x) {
        return bitset<32>(x).count();
    }
    bool canSortArray(vector<int>& nums) {
        vector<int> sortedArr = nums;
        sort(sortedArr.begin(), sortedArr.end());
        int n = nums.size();
        bool swapped;
        do {
            swapped = false;
            for (int i = 0; i < n - 1; i++) {
                if (setBitsCount(nums[i]) == setBitsCount(nums[i + 1])) {
                    if (nums[i] > nums[i + 1]) {
                        std::swap(nums[i], nums[i + 1]);
                        swapped = true;
                    }
                }
            }
        } while (swapped);
        return nums == sortedArr;
    }
};

File: 3291-find-if-array-can-be-sorted/3291-find-if-array-can-be-sorted.java
class Solution {
    public static int setBitsCount(int x) {
        return Integer.bitCount(x);
    }
    public boolean canSortArray(int[] nums) {
        int[] sortedArr = nums.clone();
        Arrays.sort(sortedArr);
        int n = nums.length;
        boolean swapped;
        do{
            swapped = false;
            for (int i = 0; i < n - 1; i++) {
                if (setBitsCount(nums[i]) == setBitsCount(nums[i + 1])) {
                    if (nums[i] > nums[i + 1]) {
                        int temp = nums[i];
                        nums[i] = nums[i + 1];
                        nums[i + 1] = temp;
                        swapped = true;
                    }
                }
            }
        }while(swapped);
        return Arrays.equals(nums, sortedArr);
    }
}